### DISCUSSION TITLE
Categorizing profiles

### DISCUSSION THREAD

**Discussion by mike-agrenius-kushner** [pkimetal/pkimetal#200] [Ideas]
Hi, 

Looking at integrating against pkimetal – a thought I had is that while the API allows polling for available linters and profiles, it doesn't offer a direct connection between the two, nor any categorization besides trying to parse the Description field in the response. 

As a FR, may I suggest adding the two following fields to the responde from the **profiles** endpoint?

1. A field describing from which linter the profile stems
2. A field describing a grouping (DV, EV, CRL, ARI, etc)

Cheers,
Mike

**Comment by robstradling** [comment]
Hi @mike-agrenius-kushner.

> 1. A field describing from which linter the profile stems

Each of pkimetal's linter integrations knows which profiles it handles, so I can use that information to programmatically populate the (array) field you're proposing here.

> 2. A field describing a grouping (DV, EV, CRL, ARI, etc)

If this would be useful to you, I'd be happy to add it.  Would you mind proposing a more complete list of groupings?

**Comment by robstradling** [comment]
> > 1. A field describing from which linter the profile stems
>
> Each of pkimetal's linter integrations knows which profiles it handles, so I can use that information to programmatically populate the (array) field you're proposing here.

Implemented by https://github.com/pkimetal/pkimetal/pull/201

Demo: https://dev.pkimet.al/profiles

**Comment by mike-agrenius-kushner** [comment]
Wow, that was quick! 

Looking at the output from 1, it actually looks like I have more than what I need – so count 2 as Not Needed.  Thanks for the help!

Whelp, we should have support in EJBCA for pkimetal in one of the coming releases – I'll send a pull request to the README once it goes live, if you'd like.

Cheers,
Mike

**Reply by robstradling** [reply]
> Whelp, we should have support in EJBCA for pkimetal in one of the coming releases – I'll send a pull request to the README once it goes live, if you'd like.

Yes please.  :-)
